applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c

```diff
@@ -593,13 +545,48 @@ int broadcast_sink_disable(void)
 int broadcast_sink_enable(le_audio_receive_cb recv_cb)
 {
 	int ret;
+	static bool initialized;
+	enum audio_channel channel;
+
+	if (initialized) {
+		LOG_WRN("Already initialized");
+		return -EALREADY;
+	}
+
+	if (recv_cb == NULL) {
+		LOG_ERR("Receive callback is NULL");
+		return -EINVAL;
+	}
+
+	receive_cb = recv_cb;
+
+	channel_assignment_get(&channel);
+
+	if (channel == AUDIO_CH_L) {
+		ret = bt_pacs_set_location(BT_AUDIO_DIR_SINK, BT_AUDIO_LOCATION_FRONT_LEFT);
+	} else {
+		ret = bt_pacs_set_location(BT_AUDIO_DIR_SINK, BT_AUDIO_LOCATION_FRONT_RIGHT);
+	}
 
-	ret = initialize(recv_cb);
 	if (ret) {
-		LOG_ERR("Failed to initialize");
+		LOG_ERR("Location set failed");
```

Comment:
This works now, but for future-proofing, I suggest making 565 a switch case. Also, if the channel for some damned reason is not equal to either `AUDIO_CH_L` or `AUDIO_CH_R` this won't be caught.

Reply:
This was just moved. Suggest this is done in a separate PR in that case.